README.org
#+title: Ox-Hugo: A carefully crafted Org exporter back-end for Hugo
#+author: Kaushal Modi
[[https://github.com/kaushalmodi/ox-hugo/actions][https://github.com/kaushalmodi/ox-hugo/actions/workflows/test.yml/badge.svg]] [[https://melpa.org/#/ox-hugo][file:https://melpa.org/packages/ox-hugo-badge.svg]] [[https://www.gnu.org/licenses/gpl-3.0][https://img.shields.io/badge/License-GPL%20v3-blue.svg]]

/If you have any questions or if you have anything interesting to
share related to ox-hugo, feel free to do so on [[https://github.com/kaushalmodi/ox-hugo/discussions][Discussions]]!/

=ox-hugo= is an Org exporter backend that exports Org to
[[https://gohugo.io/][Hugo]]-compatible Markdown ([[https://github.com/russross/blackfriday][Blackfriday]]) and also generates the
front-matter (in TOML or YAML format).

The =ox-hugo= backend extends from a /parent/ backend
=ox-blackfriday.el=. The latter is the one that primarily does the
Blackfriday-friendly Markdown content generation. The main job of
=ox-hugo= is to generate the front-matter for each exported content
file, and then append that generated Markdown to it.

There are, though, few functions that =ox-hugo.el= overrides over
those by =ox-blackfriday.el=.

-----
See the [[https://ox-hugo.scripter.co/doc/examples/][Real World Examples]] section to quickly jump to sites generated
using =ox-hugo= and their Org sources.
* Table of Contents
- [[#screenshots][Screenshots]]
- [[#documentation][Documentation]]
  - [[#source-of-the-documentation-site][Source of the Documentation site]]
- [[#demo][Demo]]
- [[#installation][Installation]]
  - [[#use-package][Use Package]]
  - [[#spacemacs][Spacemacs]]
- [[#usage][Usage]]
  - [[#before-you-export][Before you export]]
  - [[#export-bindings][Export bindings]]
  - [[#customization-options][Customization Options]]
- [[#thanks][Thanks]]

* Screenshots
Before you read further, you can see below how =ox-hugo= translates
Org to Markdown (Org on the left; exported Markdown with Hugo
front-matter on the right).
** One post per Org subtree (preferred)
[[https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/doc/static/images/one-post-per-subtree.png][https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/doc/static/images/one-post-per-subtree.png]]
- Files in above screenshot :: [[https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/test/site/content-org/screenshot-subtree-export-example.org][Org]] -> [[https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/test/site/content/writing-hugo-blog-in-org-subtree-export.md][Markdown]]
** One post per Org file
[[https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/doc/static/images/one-post-per-file.png][https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/doc/static/images/one-post-per-file.png]]
- Files in above screenshot :: [[https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/test/site/content-org/writing-hugo-blog-in-org-file-export.org][Org]] -> [[https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/test/site/content/writing-hugo-blog-in-org-file-export.md][Markdown]]
** Editorial
The preferred way to organize the posts is as Org subtrees (also the
main reason to write this package, as nothing like that was out there)
as it makes the meta-data management for Hugo front-matter pretty
effortless.

If you are a /one Org-file per post/ type of a person, that flow works
too! Just note that in this flow many of those =#+hugo_= properties
need to be managed manually.. just as one would manage the front-matter
in Markdown files --- See the Org versions in the above screenshots for
comparison.
* Documentation
=ox-hugo= uses *itself* to generate its documentation!

https://ox-hugo.scripter.co/

You can generate the same too! Simply clone this repo and do =make
doc_md=.

*Make sure you visit the above link to read more on:*
- [[https://ox-hugo.scripter.co/doc/why-ox-hugo/][Why =ox-hugo=?]]
- [[https://ox-hugo.scripter.co/doc/auto-export-on-saving/][Auto exporting to Markdown each time the Org file is saved]]
- [[https://ox-hugo.scripter.co/doc/org-capture-setup/][Using Org Capture to start a new blog post]]
- .. and many more topics and examples
** Source of the Documentation site
- [[https://raw.githubusercontent.com/kaushalmodi/ox-hugo/main/doc/ox-hugo-manual.org][Org source]]

The documentation site is published by first using =ox-hugo= to
export from Org to Markdown, and then finally =hugo=.
/So no Markdown files are committed in the =doc/content/= directory./
* Demo
[[https://github.com/kaushalmodi/ox-hugo/tree/main/test/site/content-org][Org source]] → [[https://github.com/kaushalmodi/ox-hugo/tree/main/test/site/content][=ox-hugo= Exported Markdown]] → [[https://ox-hugo.scripter.co/test/]]

The test site uses a [[https://github.com/kaushalmodi/hugo-bare-min-theme][minimal]] theme written just for debug purposes
(not extra aesthetics). The test site is designed to verify if all the
content translates from Org to Markdown as expected.

/See [[https://themes.gohugo.io/][Hugo Themes]] for examples of really good site prettification and
presentation styles./

* Installation
This package requires at least GNU Emacs *26.3* and
Org Mode *9.0*. It is available on Melpa
([[https://melpa.org/#/ox-hugo]]), and it's recommended to install this
package from [[https://melpa.org/#/getting-started][Melpa]] (​_not_ Melpa /Stable/).

You will need to /require/ the package after installing it to get the
=ox-hugo= export options in the /Org Export Dispatcher/ menu (the one
you see when you hit =C-c C-e= to initiate any export).

You can do that by adding the below to your config:
#+begin_src emacs-lisp
(with-eval-after-load 'ox
  (require 'ox-hugo))
#+end_src
** Use Package
If you use [[https://github.com/jwiegley/use-package][=use-package=]], you can do the below instead:
#+begin_src emacs-lisp
(use-package ox-hugo
  :ensure t   ;Auto-install the package from Melpa
  :pin melpa  ;`package-archives' should already have ("melpa" . "https://melpa.org/packages/")
  :after ox)
#+end_src

** Spacemacs
Spacemacs users can use =ox-hugo= by setting the variable
=org-enable-hugo-support=.

#+begin_src emacs-lisp
(setq-default dotspacemacs-configuration-layers
              '((org :variables
                  org-enable-hugo-support t)))
#+end_src

/This was verified to work on Spacemacs =develop= branch ([[https://github.com/kaushalmodi/ox-hugo/pull/440][ref]])./
* Usage
#+begin_note
Jump to the [[https://ox-hugo.scripter.co/doc/quick-start/][Quick Start]] section to quickly try
out ~ox-hugo~ with Hugo.
#+end_note
** Before you export
Before you export check that these properties are set as you need:
- HUGO_BASE_DIR :: Root directory of the source for the Hugo site. If
     this is set to =~/hugo/=, the exported Markdown files will be saved
     to =~/hugo/content/<HUGO_SECTION>/= directory[fn:-1-section_more].  By
     default, the Markdown files reside in a hierarchy under the
     =content/= directory in the site root directory ([[https://gohugo.io/content-management/organization/][ref]]).

     If you try to export without setting this property, you will get
     this error:
     #+begin_example
       user-error: It is mandatory to set the HUGO_BASE_DIR property
                   or the `org-hugo-base-dir' local variable
     #+end_example
     This property can be set by one of two ways:
  1. Setting the ~#+hugo_base_dir:~ keyword in the Org file.
  2. Setting the ~org-hugo-base-dir~ variable in a ~.dir-locals.el~ or
     File Local Variables.

- HUGO_SECTION :: The default Hugo section name for all the posts.
     See [[https://gohugo.io/content-management/sections/][here]] for more information on Hugo sections.  It is common for
     this property to be set to =posts= or =blog=.  The default value is
     set using =org-hugo-default-section-directory=. See
     [[https://ox-hugo.scripter.co/doc/hugo-section/][Hugo Section]] for details.

*Important*: If you choose to export an Org subtree as a post, you
 need to set the =EXPORT_FILE_NAME= subtree property. That property is
 used by this package to figure out where the current post starts. For
 that reason, *a subtree with =EXPORT_FILE_NAME= property cannot nest
 another subtree with that property*. If you can analogize with the
 [[https://en.wikipedia.org/wiki/Tree_(data_structure)][branch/leaf data structure terminlogy]], then the subtrees with
 =EXPORT_FILE_NAME= property need to be /leaf nodes/.

[fn:-1-section_more] The ~HUGO_SECTION~ is the bare-minimum requirement
to specify the destination path. That path can be further tweaked
using ~HUGO_BUNDLE~ key (and the associated ~EXPORT_HUGO_BUNDLE~
property), and the ~EXPORT_HUGO_SECTION_FRAG~ property (only for
/per-subtree/ exports).
** Export bindings
The common =ox-hugo= export bindings are:
*** For both one-post-per-subtree and one-post-per-file flows
- =C-c C-e H H= :: Export "What I Mean". This is same as calling the
     ~org-hugo-export-wim-to-md~ function interactively or via
     ~(org-hugo-export-wim-to-md)~ in Emacs Lisp.
  - If point is in a /valid Hugo post subtree/, export that
    subtree to a Hugo post in Markdown.

    A /valid Hugo post subtree/ is an Org subtree that has the
    =EXPORT_FILE_NAME= property set. *Note that a subtree with
    =EXPORT_FILE_NAME= property cannot nest a subtree with the same
    property set.* If you can analogize with the [[https://en.wikipedia.org/wiki/Tree_(data_structure)][branch/leaf data
    structure terminlogy]], then the subtrees with =EXPORT_FILE_NAME=
    property need to be /leaf nodes/.
  - If the file is intended to be exported as a whole (i.e. has the
    =#+title= keyword), export the whole Org file to a Hugo post in
    Markdown.
- =C-c C-e H A= :: Export *all* "What I Mean". This is same as
     executing ~(org-hugo-export-wim-to-md :all-subtrees)~ in Emacs Lisp.
  - If the Org file has one or more 'valid Hugo post subtrees', export
    them to Hugo posts in Markdown.
  - If the file is intended to be exported as a whole (i.e. no 'valid
    Hugo post subtrees' at all, and has the =#+title= keyword), export
    the whole Org file to a Hugo post in Markdown.
*** For only the one-post-per-file flow
- =C-c C-e H h= :: Export the Org file to a Hugo post in
     Markdown. This is same as calling the ~org-hugo-export-to-md~
     function interactively.

/Also see the [[https://ox-hugo.scripter.co/doc/auto-export-on-saving/][Auto Exporting]]
section./
** Customization Options
Do =M-x customize-group=, and select =org-export-hugo= to see the
available customization options for this package.
* Thanks
- Matt Price ([[https://github.com/titaniumbones][@titaniumbones]])
- Puneeth Chaganti ([[https://github.com/punchagan][@punchagan]])
- Also thanks to
  ~http://whyarethingsthewaytheyare.com/setting-up-the-blog/~ (/not
  hyperlinking the link as it is insecure --- not https/),
  ~http://www.holgerschurig.de/en/emacs-blog-from-org-to-hugo/~ (/not
  hyperlinking the link as it is insecure --- not https/) and the
  [[https://github.com/chaseadamsio/goorgeous][=goorgeous=]] project by Chase Adams ([[https://github.com/chaseadamsio][@chaseadamsio]]) for inspiration
  to start this project.


CHANGELOG.org
# This file is auto-generated by running 'make changelog' from the repo root.

* Changelog

All notable changes to this project will be documented in this file.

** *0.12.2* - <2022-06-16>

[[https://github.com/kaushalmodi/ox-hugo/compare/3c8c0bf28cf02128ceff9a972db3ebad01083953...af3d0cdf8c672be498d54bc9efdd0e40b2528602][3c8c0bf...af3d0cd]]

*** :sparkles: Features
:PROPERTIES:
:CUSTOM_ID: features-v0.12.2
:END:

- Export links to manual pages of GNU software like tar too … New custom variable `org-hugo-info-gnu-software`. It contains a list … of GNU software taken from https://www.gnu.org/software/. ([[https://github.com/kaushalmodi/ox-hugo/commit/e2da16e1752ba71198f6c7cf1a0a15c0c51f2642][e2da16e]])

*** :rocket: PR Merges
:PROPERTIES:
:CUSTOM_ID: pr-merges-v0.12.2
:END:

- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/649][649]]: … fix: Escape Hugo shortcodes in description front-matter ([[https://github.com/kaushalmodi/ox-hugo/commit/df0b12ef1aff2b322403f24006f0ec17676d3606][df0b12e]])
- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/652][652]]: … feat: Export links to manual pages of GNU software like tar too ([[https://github.com/kaushalmodi/ox-hugo/commit/2b46cd602c9692e339833a765ef6fe3acc4fa2ec][2b46cd6]])

*** :bug: Bug Fixes
:PROPERTIES:
:CUSTOM_ID: bug-fixes-v0.12.2
:END:

- Escape Hugo shortcodes in description front-matter … The description front-matter will most likely be parsed as Markdown … string using $.Page.RenderString, and we do not want literal strings … like "… ([[https://github.com/kaushalmodi/ox-hugo/commit/4f6517dfe18a08387a786d666deb326f8342ea4c][4f6517d]])
- Don't attempt to unquote empty strings … Fixes Issue # [[https://github.com/kaushalmodi/ox-hugo/issues/655][655]]. ([[https://github.com/kaushalmodi/ox-hugo/commit/af3d0cdf8c672be498d54bc9efdd0e40b2528602][af3d0cd]])

** *0.12.1* - <2022-05-31>

[[https://github.com/kaushalmodi/ox-hugo/compare/5ad69b40a6c8609f92268a76ecf4b8cf1700350b...3c8c0bf28cf02128ceff9a972db3ebad01083953][5ad69b4...3c8c0bf]]

*** :boom: Breaking
:PROPERTIES:
:CUSTOM_ID: breaking-v0.12.1
:END:

- Remove auto-insertion of Bibliography heading for org-ref cit. … Adding this feature for org-ref is a bit painful because: … 1. `org-ref` uses `citeproc-render-bib` in `org-ref-export.el`., while … th… ([[https://github.com/kaushalmodi/ox-hugo/commit/22065f719580dd6e46762dcf090957702c54dc43][22065f7]])

*** :sparkles: Features
:PROPERTIES:
:CUSTOM_ID: features-v0.12.1
:END:

- Auto-escape Hugo shortcodes in inline src blocks ([[https://github.com/kaushalmodi/ox-hugo/commit/a23ff91eb44486d571da390f2bbfed1cb2bf6e9b][a23ff91]])
- Add a commented out test case for an inline src block parsing issue … .. in upstream Org mode. … Ref: https://lists.gnu.org/r/emacs-orgmode/2022-03/msg00008.html ([[https://github.com/kaushalmodi/ox-hugo/commit/c0e762e573986ee46782ed8636ddfe05f6fdb8b6][c0e762e]])
- Use HTML code element with classes to export inline-src blocks … Fixes Issue # [[https://github.com/kaushalmodi/ox-hugo/issues/638][638]]. ([[https://github.com/kaushalmodi/ox-hugo/commit/39566c2e6ee09fc209b15e301345437dcfaf7b22][39566c2]])

*** :rocket: PR Merges
:PROPERTIES:
:CUSTOM_ID: pr-merges-v0.12.1
:END:

- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/636][636]]: … Revert "tool(ci): Disable CI runs on emacs master for now" ([[https://github.com/kaushalmodi/ox-hugo/commit/68916a64a9015a5f09ea5a9afa110a88ef0ad474][68916a6]])
- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/637][637]]: … refactor: Make the org-hugo-drawer function more readable ([[https://github.com/kaushalmodi/ox-hugo/commit/8914b75f51e3340d4a8d394f53ac2ece20e649c0][8914b75]])
- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/639][639]]: … perf: Better way to auto-inject bibliography heading ([[https://github.com/kaushalmodi/ox-hugo/commit/e63bb0952303e01db0a5af9a73540d9e45dfeea5][e63bb09]])
- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/578][578]]: … Improve inline src blocks -- they now export to HTML `code` elements with `inline-src` class and language-specific class and data attri… ([[https://github.com/kaushalmodi/ox-hugo/commit/1d4221550406b737a094aee991cdd4edc69625f9][1d42215]])
- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/643][643]]: … Fix auto menu weight calculation during "all subtrees" exports ([[https://github.com/kaushalmodi/ox-hugo/commit/d3faba4549518006fa4d7eeca657fd4142718c4b][d3faba4]])
- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/641][641]]: … fix: Change inline src block output to work around CommonMark issue ([[https://github.com/kaushalmodi/ox-hugo/commit/3070aaab5bbe297b6f90620c29171ced0bd4c439][3070aaa]])
- PR # [[https://github.com/kaushalmodi/ox-hugo/pull/646][646]]: … fix: Remove the blank line inserted between code blocks and captions ([[https://github.com/kaushalmodi/ox-hugo/commit/4cc8de4ec5f87bfa7fca97d61ae445ea0a75ae1e][4cc8de4]])

*** :white_check_mark: Issue Fixes
:PROPERTIES:
:CUSTOM_ID: issue-fixes-v0.12.1
:END:

- Revert "tool(ci): Disable CI runs on emacs master for now" … This reverts commit commit # [[https://github.com/kaushalmodi/ox-hugo/commit/288c2f577be5c1e9b0871d1c5a02b3b28346615a][288c2f5]]. … Fixes I… ([[https://github.com/kaushalmodi/ox-hugo/commit/21269f664bcf5dd5ddab6cf01d41c22821000e93][21269f6]])

*** :zap: Performance
:PROPERTIES:
:CUSTOM_ID: performance-v0.12.1
:END:

- Better way to auto-inject bibliography heading … Earlier method wasn't elegant because it relied on an empirically … derived regexp. The reason for this change was that this regexp was … falsely match… ([[https://github.com/kaushalmodi/ox-hugo/commit/5a24d8f31f04a00a0ff6364ecf0a7e2c4afc8e02][5a24d8f]])

*** :bug: Bug Fixes
:PROPERTIES:
:CUSTOM_ID: bug-fixes-v0.12.1
:END:

- Correct auto menu weight calc when batch-exporting subtrees … Fixes Issue # [[https://github.com/kaushalmodi/ox-hugo/issues/642][642]]. ([[https://github.com/kaushalmodi/ox-hugo/commit/f9c9aaf0160bb552d438cda20b7627a41dfaf125][f9c9aaf]])

example-site/elisp/setup-ox-hugo.el
;; Time-stamp: <2017-09-20 15:31:54 kmodi>

;; Setup to test ox-hugo using emacs -Q and the latest stable version of Org

;; Some sane settings
(setq-default require-final-newline t)
(setq-default indent-tabs-mode nil)

(defvar ox-hugo-elpa (let ((dir (getenv "OX_HUGO_ELPA")))
                       (unless dir
                         (setq dir (concat (temporary-file-directory) (getenv "USER") "/ox-hugo-dev/")))
                       (make-directory (file-name-as-directory dir) :parents)
                       dir))
(message "ox-hugo-elpa: %S" ox-hugo-elpa)

(let* ((bin-dir (when (and invocation-directory
                           (file-exists-p invocation-directory))
                  (file-truename invocation-directory)))
       (prefix-dir (when bin-dir
                     (replace-regexp-in-string "bin/\\'" "" bin-dir)))
       (share-dir (when prefix-dir
                    (concat prefix-dir "share/")))
       (lisp-dir-1 (when share-dir ;Possibility where the lisp dir is something like ../emacs/26.0.50/lisp/
                     (concat share-dir "emacs/"
                             ;; If `emacs-version' is x.y.z.w, remove the ".w" portion
                             ;; Though, this is not needed and also will do nothing in emacs 26+
                             ;; http://git.savannah.gnu.org/cgit/emacs.git/commit/?id=22b2207471807bda86534b4faf1a29b3a6447536
                             (replace-regexp-in-string "\\([0-9]+\\.[0-9]+\\.[0-9]+\\).*" "\\1" emacs-version)
                             "/lisp/")))
       (lisp-dir-2 (when share-dir ;Possibility where the lisp dir is something like ../emacs/25.2/lisp/
                     (concat share-dir "emacs/"
                             (replace-regexp-in-string "\\([0-9]+\\.[0-9]+\\).*" "\\1" emacs-version)
                             "/lisp/"))))
  ;; (message "setup-packages:: bin-dir: %s" bin-dir)
  ;; (message "setup-packages:: prefix-dir: %s" prefix-dir)
  ;; (message "setup-packages:: share-dir: %s" share-dir)
  ;; (message "setup-packages:: lisp-dir-1: %s" lisp-dir-1)
  ;; (message "setup-packages:: lisp-dir-2: %s" lisp-dir-2)
  (defvar my/default-lisp-directory (cond
                                     ((file-exists-p lisp-dir-1)
                                      lisp-dir-1)
                                     ((file-exists-p lisp-dir-2)
                                      lisp-dir-2)
                                     (t
                                      nil))
    "Directory containing lisp files for the Emacs installation.

This value must match the path to the lisp/ directory of the
Emacs installation.  If Emacs is installed using
--prefix=\"${PREFIX_DIR}\" this value would typically be
\"${PREFIX_DIR}/share/emacs/<VERSION>/lisp/\"."))

(defvar my/packages '(org))

(if (and (stringp ox-hugo-elpa)
         (file-exists-p ox-hugo-elpa))
    (progn
      ;; Load newer version of .el and .elc if both are available
      (setq load-prefer-newer t)

      (setq package-user-dir (format "%selpa_%s/" ox-hugo-elpa emacs-major-version))

      ;; Below require will auto-create `package-user-dir' it doesn't exist.
      (require 'package)

      ;; Adding Melpa is not needed for now.
      ;; (let* ((no-ssl (and (memq system-type '(windows-nt ms-dos))
      ;;                     (not (gnutls-available-p))))
      ;;        (url (concat (if no-ssl "http" "https") "://melpa.org/packages/")))
      ;;   (add-to-list 'package-archives (cons "melpa" url) :append))
      (add-to-list 'package-archives '("org" . "http://orgmode.org/elpa/") :append) ;For latest `org'

      (defvar ox-hugo-git-root (progn
                                 (require 'vc-git)
                                 (vc-git-root "../Makefile")))
      (message "ox-hugo-git-root: %S" ox-hugo-git-root)
      (add-to-list 'load-path ox-hugo-git-root)

      ;; Load emacs packages and activate them.
      ;; Don't delete this line.
      (package-initialize)
      ;; `package-initialize' call is required before any of the below
      ;; can happen.

      (defvar my/missing-packages '()
        "List populated at each startup that contains the list of packages that need
to be installed.")

      (dolist (p my/packages)
        (unless (package-installed-p p)
          (add-to-list 'my/missing-packages p)))

      (when my/missing-packages
        (message "Emacs is now refreshing its package database...")
        (package-refresh-contents)
        ;; Install the missing packages
        (dolist (p my/missing-packages)
          (message "Installing `%s' .." p)
          (package-install p))
        (setq my/missing-packages '())))
  (error "The environment variable OX_HUGO_ELPA needs to be set"))

;; Remove Org that ships with Emacs from the `load-path'.
(with-eval-after-load 'package
  (when (stringp my/default-lisp-directory)
    (dolist (path load-path)
      (when (string-match-p (expand-file-name "org" my/default-lisp-directory) path)
        (setq load-path (delete path load-path))))))

;; (message "`load-path': %S" load-path)
;; (message "`load-path' Shadows:")
;; (message (list-load-path-shadows :stringp))

(require 'ox-hugo)
(defun org-hugo-export-all-subtrees-to-md ()
  (org-hugo-export-subtree-to-md :all-subtrees))

(with-eval-after-load 'org
  ;; Allow multiple line Org emphasis markup
  ;; http://emacs.stackexchange.com/a/13828/115
  (setcar (nthcdr 4 org-emphasis-regexp-components) 20) ;Up to 20 lines, default is just 1
  ;; Below is needed to apply the modified `org-emphasis-regexp-components'
  ;; settings from above.
  (org-set-emph-re 'org-emphasis-regexp-components org-emphasis-regexp-components)

  ;; Prevent prompts like:
  ;;   Non-existent agenda file
  (defun org-check-agenda-file (file)))

(with-eval-after-load 'ox
  (setq org-export-with-sub-superscripts '{}))

;; Wed Sep 20 13:37:06 EDT 2017 - kmodi
;; Below does not get applies when running emacs --batch.. need to
;; figure out a solution.
(custom-set-variables
 '(safe-local-variable-values
   (quote
    ((org-hugo-footer . "

[//]: # \"Exported with love from a post written in Org mode\"
[//]: # \"- https://github.com/kaushalmodi/ox-hugo\"")))))
